src/ImageSharp/Formats/Bmp/BmpImageFormatDetector.cs

```diff
@@ -14,7 +15,7 @@ public sealed class BmpImageFormatDetector : IImageFormatDetector
     public int HeaderSize => 2;
 
     /// <inheritdoc/>
-    public IImageFormat DetectFormat(ReadOnlySpan<byte> header)
+    public IImageFormat? DetectFormat(ReadOnlySpan<byte> header)
```

Comment:
I cannot decide what the best pattern for nullable return types should be. I feel like anything that returns a nullable type should probably use a `TryXX(in, out..)` pattern (though I still haven't found a nice async pattern for that)

Reply:
> I feel like anything that returns a nullable type should probably use a TryXX(in, out..)

yeah. i was trying to do this in the least intrusive way

> though I still haven't found a nice async pattern for that

neither have I

Reply:
happy to add some TryXX overloads. but i would prefer to do those in a new PR